### PR TITLE
New Grid Layout

### PR DESCRIPTION
**Pull request by alperozturk96**
### Light Mode Idle

| Before  | After |
| ------------- | ------------- |
| before_light_idle | light_mode_idle |

### Light Mode Multi Select

| Before  | After |
| ------------- | ------------- |
| before_light_multi_select | light_mode_multi_select |

### Dark Mode Idle

| Before  | After |
| ------------- | ------------- |
| before_dark_idle | dark_mode_idle |

### Dark Mode Multi Select

| Before  | After |
| ------------- | ------------- |
| before_dark_mode_multi_select | dark_mode_multi_select |

### More Actions In Grid Layout

more_actions_bottom_sheet

@nextcloud/designers What do you think about this implementation?

- [x] Tests written, or not not needed

### CODE REVIEW

**Review by ZetaTom — COMMENTED** [review]
Generally this seems to work well.

On my test device, some files were lacking their respective title. Notice the missing filenames on `Nextcloud.png` and `Nextcloud intro.mp4`. This also happens on `master`, I'm not sure, whether this is intended.

image

**Comment by ZetaTom** [comment]
Another thing I've noticed is that, the bottom sheet is displayed without the appropriate [scrim](https://m3.material.io/components/bottom-sheets/specs) when opening the overflow menu of any given file. However, this behaviour is also present on `master`.

**Comment by alperozturk96** [comment]
> Generally this seems to work well.
> 
> On my test device, some files were lacking their respective title. Notice the missing filenames on `Nextcloud.png` and `Nextcloud intro.mp4`. This also happens on `master`, I'm not sure, whether this is intended.
> 
> image

I didn't add titles to the media files because I think it was left like this on purpose. But we can ask the design team. I think the titles should be in the media files.  @nextcloud/designers

**Comment by alperozturk96** [comment]
> Another thing I've noticed is that, the bottom sheet is displayed without the appropriate [scrim](https://m3.material.io/components/bottom-sheets/specs) when opening the overflow menu of any given file. However, this behaviour is also present on `master`.

It's same in master so not related with this PR. However Scrim is available for only Modal Bottom Sheet not for Standard one. Couple of little details are missing due to multiple things.

- Some components still using AlertDialog and override appearance from themes.xml
- Showing BottomSheet or Dialog done by separate classes actually it's not needed. e.g SendFilesDialog, RenamePublicShareDialogFragment. No need to use DialogFragment or BottomSheetDialogFragment. You can directly show components from activity and fragment without creating another fragment and therefore you will get default behaviour.
- Mandatory use of viewThemeUtils makes it difficult to use default components.

Current implementation of M3 uses latest MDC library and themes.xml values. Trying to solve little details like scrim may take a lot of time and the benefit will be minimal for us. I suggest you open details about components in a new design issue

File: app/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -23,6 +23,8 @@
     <color name="white">#ffffff</color>
     <color name="white_helper_text">#B3FFFFFF</color>
     <color name="text_color">#333333</color>
+    <color name="grid_icon_color">#303034</color>
+    <color name="grid_background_color">#E9E8EB</color>
```
[the remark this answers is not in the document]
Comment (alperozturk96) [reply app/src/main/res/values/colors.xml:27]: Nope they can't be unified. Grid files features layout must have different background color. I will rename it

**Comment by alperozturk96** [comment]
@tobiasKaminsky @AndyScherzinger I think we should also display file name for media files such as image, video. WDYT?

**Comment by AndyScherzinger** [comment]
Well, depends on the preview size for images. If they come in a smaller size as reflected in your PR, than there is space to show the name. However for pure image folders this will take away the "mosaic" effect of pictures. So I'd leave it to the designers' decision.

**Comment by alperozturk96** [comment]
> Well, depends on the preview size for images. If they come in a smaller size as reflected in your PR, than there is space to show the name. However for pure image folders this will take away the "mosaic" effect of pictures. So I'd leave it to the designers' decision.

Using different image sizes in file list grid adapter not a good idea. On master folders always have same size and media also always have same size. Media ones slightly bigger than folder. In my PR both have same size therefore we will have consistent layout and better alignment.

**Comment by nimishavijay** [comment]
> I didn't add titles to the media files because I think it was left like this on purpose.

Good point, in many cases we opt for not showing the file names for media because they are not providing much information. But in this case it makes sense to show the file names. 

> However for pure image folders this will take away the "mosaic" effect of pictures. 

This is likely more of an edge case as chances are people are using the Photos app to view their photos like this, and in the Files app they are doing some operations like moving, etc. So I am okay with showing a smaller thumbnail and file name.  

Feedback on the design itself:
The rounded corners are great! The gap between the selected files is also nice, helps to really see what is selected.

- [ ] Regarding the star (favorite icon): 
  - [ ] The size seems very small, we can increase it by 2-4px at least :) 
  - [ ] Generally we have a pattern of showing the favorite icon in the top right corner in all apps (Files, Talk, Mail, etc). What do you think about keeping the position of the favourite and sharing icon in the top right like how it was before, but moving it towards slightly towards the folder? So 4px down and 4px to the left? 
  - [ ] In dark mode the half transparent background should be dark as well

- [ ] When a file is multiselected, there is more spacing on top than at the bottom. Ideally it should be centered, so decrease the space on top and increase in the bottom :) 
image
- [ ] When files are multiselected, it would be confusing to show the 3 dot menu for each file as well as in the top bar. I would recommend that when more than one file is selected, the 3 dot menu is hidden.
- [ ] When a file is multiselected, the size of the checkbox is also pretty small. I suggest we can also keep the position and size of the checkbox in the top left corner and increase the size.

**Comment by AndyScherzinger** [comment]
> What do you think about keeping the position of the favourite and sharing icon in the top right like how it was before, but moving it towards slightly towards the folder? So 4px down and 4px to the left?

@nimishavijay this is a result of another discussion in some other place between @alperozturk96, me and @jancborchardt to move to this way of displaying it on Android since this is also how Google Drive does it and helps us with potential color issues especially on images,.

**Comment by nimishavijay** [comment]
@AndyScherzinger ah understood, I would only suggest that the icons in that container be a bit bigger :)
However, for the checkbox I would strongly recommend it is in the top left corner, as it seems to get lost with the rest of the icons there.

**Comment by alperozturk96** [comment]
@AndyScherzinger @nimishavijay 

Here is the latest iteration of new grid layout...

https://github.com/nextcloud/android/assets/67455295/7d7573ee-ce19-4cd4-ac3f-a12396ea0ecc

Filenames for Media

Screenshot 2023-12-14 at 13 46 58

**Comment by AndyScherzinger** [comment]
looks good to me @alperozturk96 👍

**Comment by nimishavijay** [comment]
Looks good to me as well! 
One note: I see the little bubble on the bottom right corner when an item is selected. If an item is not starred or shared then this bubble should not appear when the item is selected :)

**Comment by alperozturk96** [comment]
> Looks good to me as well! One note: I see the little bubble on the bottom right corner when an item is selected. If an item is not starred or shared then this bubble should not appear when the item is selected :)

Yes I am aware of it. We have that problem on master as well. I will try to fix that bug, thanks 👍

**Comment by alperozturk96** [comment]
@AndyScherzinger  @ZetaTom @nimishavijay 

### Little Bubble Fix
https://github.com/nextcloud/android/assets/67455295/658dfab1-4e0e-4a0f-8b52-8acc3d18d9f4

### Move or Copy File Feature Layout Visiblity Fix
https://github.com/nextcloud/android/assets/67455295/cf117947-c947-45fc-95af-cba5e1febcce

Could you approve PR if it's okay?

**Review by JonasMayerDev — APPROVED** [review]
Works for me

**Pull request merged**